packages/mobile/src/pincode/authentication.ts

```diff
@@ -134,25 +137,42 @@ export async function getPassword(
 ) {
   while (passwordLock) {
     await sleep(100)
+    if (lastPassword) {
+      return lastPassword
+    }
+    if (lastError) {
+      throw lastError
+    }
```

Comment:
This can be wrong if the the call to `getPassword` passes a different account in the 500ms window.
Now it's probably not something we do now.

Reply:
It's just I saw we started adding another Geth account for CIP8 to encrypt/decrypt/sign using the data encryption key. Not sure how we want to handle that. 
Would be nice to chat together about this (@isabellewei, @tarikbellamine, @nategraf).